cognite/client/data_classes/data_modeling/data_types.py

```diff
@@ -48,78 +46,98 @@ def as_tuple(self) -> tuple[str, str]:
 @dataclass
 class PropertyType(ABC):
     _type: ClassVar[str]
+    is_list: bool = False
 
     def dump(self, camel_case: bool = True) -> dict[str, Any]:
         output = asdict(self)
         output["type"] = self._type
         for key in list(output.keys()):
             if output[key] is None:
                 output.pop(key)
-        output = rename_and_exclude_keys(output, aliases=_PROPERTY_ALIAS_INV)
+        output = rename_and_exclude_keys(output, aliases=_PROPERTY_ALIAS)
         return convert_all_keys_recursive(output, camel_case)
 
+    @staticmethod
+    def __load_unit_ref(data: dict) -> UnitReference | None:
+        unit: UnitReference | None = None
+        if (unit_raw := data.get("unit")) and isinstance(unit_raw, dict):
+            unit = UnitReference.load(unit_raw)
+        return unit
+
     @classmethod
     def load(cls, data: dict) -> Self:
-        if "type" not in data:
-            raise ValueError("Property types are required to have a type")
         type_ = data["type"]
-        data = convert_all_keys_to_snake_case(rename_and_exclude_keys(data, aliases=_PROPERTY_ALIAS, exclude={"type"}))
-
-        if type_cls := _TYPE_LOOKUP.get(type_):
-            if issubclass(type_cls, LoadablePropertyType):
-                return cast(Self, type_cls.load(data))
-            try:
-                return cast(Self, type_cls(**data))
-            except TypeError:
-                not_supported = set(data).difference(inspect.signature(type_cls).parameters) - {"type"}
-                logger.warning(
-                    f"For '{type_cls.__name__}', the following properties are not yet supported in the SDK (ignored): "
-                    f"{not_supported}. Try updating to the latest SDK version, or create an issue on Github!"
-                )
-                return cast(Self, type_cls(**rename_and_exclude_keys(data, exclude=not_supported)))
-
-        raise ValueError(f"Invalid type {type_}.")
+        obj: Any
```

Comment:
Nice cleanup

Reply:
Why change O(1) lookup via _TYPE_LOOKUP to O(N) if-else search? Seems like a step backwards

Reply:
I don't agree. This is not on a performance-critical path, so I prefer type safety and simplicity.

Reply:
Type safety?

Reply:
Simplicity 😅 
<img width="351" alt="image" src="https://github.com/cognitedata/cognite-sdk-python/assets/8521241/0790fb7d-8026-47ea-b1f5-8cf2070020f4">